netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -350,6 +351,15 @@ private void createStream(CreateStreamCommand command, final ChannelPromise prom
     final NettyClientStream stream = command.stream();
     final Http2Headers headers = command.headers();
     stream.id(streamId);
+
+    if (connection().goAwayReceived() && streamId > connection().local().lastStreamKnownByPeer()) {
+      // A GOAWAY was already received and this stream will never be known by the remote endpoint.
```

Comment:
Even if there is a streamId we could theoretically use, the spec says we mustn't create any more streams (Section 6.8):
> Receivers of a GOAWAY frame MUST NOT open additional streams on the connection, although a new connection can be established for new streams.